worlds/jakanddaxter/Regions.py

```diff
@@ -0,0 +1,210 @@
+from BaseClasses import MultiWorld, CollectionState, ItemClassification
+from Options import OptionError
+from .JakAndDaxterOptions import (JakAndDaxterOptions,
+                                  EnableMoveRandomizer,
+                                  EnableOrbsanity,
+                                  CompletionCondition)
+from .Items import (JakAndDaxterItem,
+                    item_table,
+                    move_item_table)
+from .Rules import can_reach_orbs
+from .locs import (CellLocations as Cells,
+                   ScoutLocations as Scouts)
+from .regs.RegionBase import JakAndDaxterRegion
+from .regs import (GeyserRockRegions as GeyserRock,
+                   SandoverVillageRegions as SandoverVillage,
+                   ForbiddenJungleRegions as ForbiddenJungle,
+                   SentinelBeachRegions as SentinelBeach,
+                   MistyIslandRegions as MistyIsland,
+                   FireCanyonRegions as FireCanyon,
+                   RockVillageRegions as RockVillage,
+                   PrecursorBasinRegions as PrecursorBasin,
+                   LostPrecursorCityRegions as LostPrecursorCity,
+                   BoggySwampRegions as BoggySwamp,
+                   MountainPassRegions as MountainPass,
+                   VolcanicCraterRegions as VolcanicCrater,
+                   SpiderCaveRegions as SpiderCave,
+                   SnowyMountainRegions as SnowyMountain,
+                   LavaTubeRegions as LavaTube,
+                   GolAndMaiasCitadelRegions as GolAndMaiasCitadel)
+
+
+def create_regions(multiworld: MultiWorld, options: JakAndDaxterOptions, player: int):
+
+    # Always start with Menu.
+    menu = JakAndDaxterRegion("Menu", player, multiworld)
+    multiworld.regions.append(menu)
+
+    # Build the special "Free 7 Scout Flies" Region. This is a virtual region always accessible to Menu.
+    # The Locations within are automatically checked when you receive the 7th scout fly for the corresponding cell.
+    free7 = JakAndDaxterRegion("'Free 7 Scout Flies' Power Cells", player, multiworld)
+    free7.add_cell_locations(Cells.loc7SF_cellTable.keys())
+    for scout_fly_cell in free7.locations:
+
+        # Translate from Cell AP ID to Scout AP ID using game ID as an intermediary.
+        scout_fly_id = Scouts.to_ap_id(Cells.to_game_id(scout_fly_cell.address))
```

Comment:
I didn't check that this is actually ensured, but you can do this if it can't be `None` (and import cast, of course)
```suggestion
        scout_fly_id = Scouts.to_ap_id(Cells.to_game_id(cast(int, scout_fly_cell.address)))
```

Reply:
Sorry, not sure I understand what `cast` is accomplishing here?

Reply:
`scout_fly_cell.address` is allowed to be `None` by your typing. But I'm assuming it _can't_ be None here for some reason

Reply:
Oh, I thought `free7.locations` would be a collection of Location objects.

Reply:
Right, and Locations can have an address of `None`, which is used for events. But, given that this code works and isn't throwing errors from handing `None` to the function, I assume these aren't using `None`